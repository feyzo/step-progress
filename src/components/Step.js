import React from 'react';
import styled from 'styled-components';
import Circle from './Circle';

const colors = {
  normal: '#d3d3d3',
  complete: '#3f00d2'
};

const StepContainer = styled.div`
  flex: ${(props) => props.showLine ? 1 : 0 };
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-top: 30px;
`;

const Button = styled.button`
  position: relative;
  display: flex;
  color: ${(props) => props.color };
  background: none;
  border: none;
  margin: 0;
  padding: 0;
  outline: 0;
  cursor: ${(props) => props.disabled ? 'not-allowed' : 'pointer' };

  circle, span {
    transition: fill, stroke 300ms linear;
  }

  &:enabled:hover circle {
    fill: ${(props) => props.complete ? colors.normal : colors.complete };
  }

  &:enabled:hover span {
    color: ${colors.complete};
  }
`;

const Label = styled.span`
  position: absolute;
  top: -30px;
  text-align: center;
  width: 80px;
  left: 50%;
  margin-left: -40px;
  font-weight: bold;
  font-size: 16px;
  color: ${(props) => props.color};
`;

const Line = styled.div`
  flex: 1;
  height: 5px;
  background-color: ${(props) => props.stroke};
  margin: 0 -2px;
`;

function Step(props) {
  const {
    name,
    complete,
    showLine,
    completeLine,
    onClick,
    enabled
  } = props;
  const stroke = complete ? colors.complete : colors.normal;
  const lineStroke = completeLine ? colors.complete : colors.normal;

  return (
    <StepContainer showLine={showLine}>
      <Button
        color={stroke}
        onClick={onClick}
        disabled={!enabled}
        type="button">
        <Label color={stroke}>{name}</Label>
        <Circle stroke={ stroke } />
      </Button>
      { showLine && (<Line stroke={ lineStroke } />)}
    </StepContainer>
  );
}

export default Step;

