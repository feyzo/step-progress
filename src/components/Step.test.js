import React from 'react';
import renderer from 'react-test-renderer';

import Step from './Step';

jest.mock('./Circle', () => 'Circle');

test('Renders a step with circle', () => {
  const component = renderer.create(
    <Step
      name={ 'test' }
      complete={ 'true' }
      showLine={ true }
      enabled={ true }
      completeLine={ false }
      key={ 'test' }
      onClick={ jest.fn() }
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});