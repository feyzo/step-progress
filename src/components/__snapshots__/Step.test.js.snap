exports[`test Renders a step with circle 1`] = `
<div
  className="fTyqPx">
  <button
    className="kdzgLC"
    color="#3f00d2"
    disabled={false}
    onClick={[Function]}
    type="button">
    <span
      className="gTEnhL"
      color="#3f00d2">
      test
    </span>
    <Circle
      stroke="#3f00d2" />
  </button>
  <div
    className="hylbHF" />
</div>
`;
